addons/website_customer/controllers/main.py

```diff
@@ -15,47 +15,62 @@ class WebsiteCustomer(http.Controller):
     @http.route([
         '/customers',
         '/customers/page/<int:page>',
-        '/customers/country/<int:country_id>',
-        '/customers/country/<country_name>-<int:country_id>',
-        '/customers/country/<int:country_id>/page/<int:page>',
-        '/customers/country/<country_name>-<int:country_id>/page/<int:page>',
-        '/customers/tag/<tag_id>',
-        '/customers/tag/<tag_id>/page/<int:page>',
-        '/customers/tag/<tag_id>/country/<int:country_id>',
-        '/customers/tag/<tag_id>/country/<country_name>-<int:country_id>',
-        '/customers/tag/<tag_id>/country/<int:country_id>/page/<int:page>',
-        '/customers/tag/<tag_id>/country/<country_name>-<int:country_id>/page/<int:page>',
+        '/customers/country/<model("res.country"):country>',
+        '/customers/country/<model("res.country"):country>/page/<int:page>',
+        '/customers/industry/<model("res.partner.industry"):industry>',
+        '/customers/industry/<model("res.partner.industry"):industry>/page/<int:page>',
+        '/customers/industry/<model("res.partner.industry"):industry>/country/<model("res.country"):country>',
+        '/customers/industry/<model("res.partner.industry"):industry>/country/<model("res.country"):country>/page/<int:page>',
     ], type='http', auth="public", website=True)
-    def customers(self, country_id=0, page=0, country_name='', tag_id=0, **post):
-        Country = request.env['res.country']
+    def customers(self, country=None, industry=None, page=0, **post):
```

Comment:
:+1: for pushing the tags into the query string. It will break a few old URLs, but such an explosion of routes would be unmaintainable.